src/parser.rs

```diff
@@ -27,164 +29,149 @@ impl<'a, T: Read> Parser<T> {
     /// values you can call this multiple times.  If the reader is not yet
     /// ready this will block.
     pub fn parse_value(&mut self) -> RedisResult<Value> {
-        let b = try!(self.read_byte());
-        match b as char {
-            '+' => self.parse_status(),
-            ':' => self.parse_int(),
-            '$' => self.parse_data(),
-            '*' => self.parse_bulk(),
-            '-' => self.parse_error(),
-            _ => fail!((ErrorKind::ResponseError, "Invalid response when parsing value")),
+        let mut buf = [0; 4];
```

Comment:
I need to figure out the best way to read from the socket. This is just for testing locally now.